pkg/kubelet/qos/util/qos.go

```diff
@@ -48,22 +45,48 @@ func isResourceBestEffort(container *api.Container, resource api.ResourceName) b
 }
 
 // GetPodQos returns the QoS class of a pod.
-// The QoS class of a pod is the lowest QoS class for each resource in each container.
+// A pod is besteffort if none of its containers have specified any requests or limits.
+// A pod is guaranteed only when requests and limits are specified for all the containers and they are equal.
+// A pod is burstable if limits and requests do not match across all containers.
 func GetPodQos(pod *api.Pod) string {
-	qosValues := sets.NewString()
+	requests := map[api.ResourceName]int64{}
+	limits := map[api.ResourceName]int64{}
+	isGuaranteed := true
 	for _, container := range pod.Spec.Containers {
-		qosPerResource := GetQoS(&container)
-		for _, qosValue := range qosPerResource {
-			qosValues.Insert(qosValue)
+		// process requests
+		for name, quantity := range container.Resources.Requests {
+			if quantity.Value() > 0 {
+				requests[name] += quantity.Value()
```

Comment:
this is prone to math problems.

i have utilities that do this more precisely.

https://github.com/kubernetes/kubernetes/blob/master/pkg/quota/resources.go#L65

I am fine moving those utilities to a different package, but generally, we should not add quantity like this because it loses precision.